module/platform/webots/Webots/src/Webots.cpp

```diff
@@ -301,13 +306,29 @@ namespace module::platform::webots {
             // Service the watchdog
             emit<Scope::WATCHDOG>(ServiceWatchdog<Webots>());
 
-            // Tick the clock forward
-            Clock::tick();
+            // Deal with time
+
+            // Save our previous deltas
+            const uint32_t prev_sim_delta  = this->sim_delta;
+            const uint32_t prev_real_delta = this->real_delta;
+
+            // Update our current deltas
+            this->sim_delta  = msg.time - this->current_sim_time;
+            this->real_delta = msg.real_time - this->current_real_time;
+
+            // Calculate our custom rtf - the ratio of the past two sim deltas and the past two real time deltas
+            utility::clock::custom_rtf = static_cast<double>(this->sim_delta + prev_sim_delta)
+                                         / static_cast<double>(this->real_delta + prev_real_delta);
+
+            // Update our current times
+            this->current_sim_time  = msg.time;
+            this->current_real_time = msg.real_time;
         });
 
+        // TODO(KipHamiltons) why every 10 ms? That seems a little random?
```

Comment:
It is completely arbitrary. It should be once per message received/time step. Maybe send it when they send us a message?

Reply:
I suppose that would be correct. That avoids the potential race condition completely as well

Reply:
It does not prevent the race condition.

Reply:
I think it would. at the end of the `on<Trigger<ServoTargets>, With<DarwinSensors>>` where we do the `std::move`, it could emit the `ActuatorRequest` instead. Then, we would change this to be `on<Trigger<ActuatorRequest>>` instead of periodically. Then we wouldn't need a persisent `to_send_x` at all.